test/sequential/test-debugger-preserve-breaks.js

```diff
@@ -9,54 +9,42 @@ const startCLI = require('../common/debugger');
 const assert = require('assert');
 const path = require('path');
 
+const scriptFullPath = fixtures.path('debugger', 'three-lines.js');
+const script = path.relative(process.cwd(), scriptFullPath);
+
 // Run after quit.
-{
-  const scriptFullPath = fixtures.path('debugger', 'three-lines.js');
-  const script = path.relative(process.cwd(), scriptFullPath);
+const runTest = async () => {
   const cli = startCLI([script]);
+  try {
+    await cli.waitForInitialBreak();
+    await cli.waitForPrompt();
+    await cli.command('breakpoints');
+    assert.match(cli.output, /No breakpoints yet/);
+    await cli.command('sb(2)');
+    await cli.command('sb(3)');
+    await cli.command('breakpoints');
+    assert.ok(cli.output.includes(`#0 ${script}:2`));
+    assert.ok(cli.output.includes(`#1 ${script}:3`));
+    await cli.stepCommand('c'); // hit line 2
+    await cli.stepCommand('c'); // hit line 3
+    assert.deepStrictEqual(cli.breakInfo, { filename: script, line: 3 });
+    await cli.command('restart');
+    await cli.waitForInitialBreak();
+    assert.deepStrictEqual(cli.breakInfo, { filename: script, line: 1 });
+    await cli.stepCommand('c');
+    assert.deepStrictEqual(cli.breakInfo, { filename: script, line: 2 });
+    await cli.stepCommand('c');
+    assert.deepStrictEqual(cli.breakInfo, { filename: script, line: 3 });
+    await cli.command('breakpoints');
+    const msg = `SCRIPT: ${script}, OUTPUT: ${cli.output}`;
+    assert.ok(cli.output.includes(`#0 ${script}:2`), msg);
+    assert.ok(cli.output.includes(`#1 ${script}:3`), msg);
 
-  function onFatal(error) {
+    await cli.quit();
+  } catch (error) {
```

Comment:
A cleaner way to do this would be with a try/finally:

```js
try {
  // your code
} finally {
  await cli.quit()
}
```

You don't need to re-throw the error or duplicate the `cli.quit()` that way.